dll/win32/shell32/wine/shell32_main.c

```diff
@@ -386,17 +408,12 @@ BOOL SHELL_IsShortcut(LPCITEMIDLIST pidlLast)
     BOOL ret = FALSE;
 
     if (_ILGetExtension(pidlLast, szTemp, _countof(szTemp)) &&
-        HCR_MapTypeToValueW(szTemp, szTemp, _countof(szTemp), TRUE))
+        SUCCEEDED(HCR_GetProgIdKeyOfExtension(szTemp, &keyCls, FALSE)))
     {
-        if (ERROR_SUCCESS == RegOpenKeyExW(HKEY_CLASSES_ROOT, szTemp, 0, KEY_QUERY_VALUE, &keyCls))
-        {
-            if (ERROR_SUCCESS == RegQueryValueExW(keyCls, L"IsShortcut", NULL, NULL, NULL, NULL))
-                ret = TRUE;
-
-            RegCloseKey(keyCls);
-        }
+        if (ERROR_SUCCESS == RegQueryValueExW(keyCls, L"IsShortcut", NULL, NULL, NULL, NULL))
```

Comment:
```suggestion
        if (RegQueryValueExW(keyCls, L"IsShortcut", NULL, NULL, NULL, NULL) == ERROR_SUCCESS)
```
since we're at it. I wonder whether we could just do: `ret = (RegQueryValueExW(...) == ERROR_SUCCESS);`